gen/field.go

```diff
@@ -53,12 +57,27 @@ func (f fieldGroupGenerator) DefineStruct(g Generator) error {
 		`type <.Name> struct {
 			<range .Fields>
 				<if .Required>
-					<goCase .Name> <typeReference .Type>
+					<goCase .Name> <typeReference .Type> <tag .>
 				<else>
-					<goCase .Name> <typeReferencePtr .Type>
+					<goCase .Name> <typeReferencePtr .Type> <tag .>
 				<end>
 			<end>
-		}`, f)
+		}`,
+		f,
+		TemplateFunc("tag", func(f *compile.FieldSpec) string {
+			// We want to add omitempty if the field is an optional struct or
+			// primitive to redure "null" noise. We won't add omitempty for
+			// optional collections because omitempty doesn't differentiate
+			// between nil and empty collections.
+
+			if (isStructType(f.Type) || isPrimitiveType(f.Type)) && !f.Required {
+				return fmt.Sprintf("`json:\"%s,omitempty\"`", f.Name)
```

Comment:
(no action for now)
We may want "plugins" for tag generation in future, since there's also "db" tags, people may want custom "json" tags etc. I imagine we'd want some way to get a list of tags, probably like `map[string]string{"json": "name,omitempty"}` that we then convert to a single tag.